apps/application-system/api/src/app/modules/application/application.service.ts

```diff
@@ -84,6 +83,31 @@ export class ApplicationService {
     })
   }
 
+  async findAllDueToBePruned(): Promise<Application[]> {
+    return this.applicationModel.findAll({
```

Comment:
This query might be quite heavy (with more applications) and large (in the first run).

Heavy:
* We should tweak the columns/indices for this so we're not performing a full table scan over all applications ever submitted.
* Maybe have an index on pruneAt, but then it still always processes all pruned applications from the start.
* We could either null pruneAt after pruning successfully (not ideal?) or add a new column indicating that the application is pruned and have a multicolumn index on `(pruned, pruneAt)`, even a partial index `WHERE (pruned = FALSE AND pruneAt IS NOT NULL)`.
* If we add a `pruned` column, then we should not filter on the other fields.

Large:
* Only select the id and attachments fields?
* Should remove the order statement, unless needed for some reason?

Reply:
Yes the pruned field was my thought also. When pruneAt = null it was my understanding that this meant that the applictation should not be pruned at all.